packages/ui/src/views/project/project-labels/project-labels-list.tsx

```diff
@@ -23,29 +47,51 @@ export const ProjectLabelsListView: React.FC<ProjectLabelPageProps> = ({
           Labels
         </Text>
         <Spacer size={6} />
-        <ListActions.Root>
-          <ListActions.Left>
-            <SearchBox.Root
-              width="full"
-              className="max-w-96"
-              // value={searchInput || ''}
-              // handleChange={handleInputChange}
-              placeholder={t('views:repos.search')}
-            />
-          </ListActions.Left>
-          <ListActions.Right>
-            <Button variant="default" onClick={openCreateLabelDialog}>
-              New label
-            </Button>
-          </ListActions.Right>
-        </ListActions.Root>
+        {(!!spaceLabels.length || (!spaceLabels.length && isDirtyList)) && (
+          <>
+            <ListActions.Root>
+              <ListActions.Left>
+                <SearchBox.Root
+                  width="full"
+                  className="max-w-96"
+                  value={searchInput || ''}
+                  handleChange={handleInputChange}
+                  placeholder={t('views:repos.search')}
+                />
+              </ListActions.Left>
+              <ListActions.Right>
+                <Button variant="default" onClick={openCreateLabelDialog}>
+                  New label
+                </Button>
+              </ListActions.Right>
+            </ListActions.Root>
+          </>
+        )}
         <Spacer size={5} />
-        <LabelsListView
-          labels={labels}
-          createdIn={createdIn}
-          handleDeleteLabel={handleDeleteLabel}
-          handleEditLabel={handleEditLabel}
-          useTranslationStore={useTranslationStore}
+        {isLoadingSpaceLabels ? (
+          <SkeletonList />
+        ) : (
+          <LabelsListView
```

Comment:
i'm curious why the SkeletonList is outside the ListView, but the NoData state is inside..